### PR TITLE
Use smaller and compressed varients of buttons and form components (#7304)

### PR DESCRIPTION
**Pull request by AMoo-Miki**
cherry picked 6f28170af93d842a0567c74bbabfe6ba9e5314c0 from #7304

## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: test/functional/apps/dashboard/dashboard_saved_query.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ export default function ({ getService, getPageObjects }) {
         await savedQueryManagementComponent.openSavedQueryManagementComponent();
         const descriptionText = await testSubjects.getVisibleText('saved-query-management-popover');
         expect(descriptionText).to.eql(
-          'SAVED QUERIES\nThere are no saved queries. Save query text and filters that you want to use again.\nSave current query'
+          'Saved Queries\nThere are no saved queries. Save query text and filters that you want to use again.\nSave current query'
```
Comment (virajsanghvi) [inline test/functional/apps/dashboard/dashboard_saved_query.js:61]: Intentional change?

**Review by virajsanghvi — APPROVED** [review]

**Comment by virajsanghvi** [comment]
Approved assuming unrelated failing checks

**Review by LDrago27 — APPROVED** [review]

File: test/functional/apps/dashboard/dashboard_saved_query.js (hunk 1 above)
Comment (AMoo-Miki) [reply test/functional/apps/dashboard/dashboard_saved_query.js:61]: This is the side effect of https://github.com/opensearch-project/oui/pull/1283

**Comment by yubonluo** [comment]
Hi @AMoo-Miki, when will this pr back port to 2.16?

**Pull request merged**